src/System.Private.CoreLib/src/System/Reflection/RuntimeAssembly.cs

```diff
@@ -643,15 +643,13 @@ public override Assembly GetSatelliteAssembly(CultureInfo culture, Version versi
             if (culture == null)
                 throw new ArgumentNullException(nameof(culture));
 
-            string name = GetSimpleName() + ".resources";
-            return InternalGetSatelliteAssembly(name, culture, version, true);
+            return InternalGetSatelliteAssembly(culture, version, true);
```

Comment:
I moved the name generation to `InternalGetSatelliteAssembly` to avoid unnecessarily duplicating it in `ManifestBasedResourceGroveler`.